informer/src/handler/subscription.rs

```diff
@@ -14,28 +14,41 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use crate::{ColdEvents, EventTags, Events, Params, Sink};
+use crate::{ColdEvents, EventTags, Events, Params, Sink, SubscriptionId};
 use jsonrpc_core::futures::Future;
+use std::sync::atomic::AtomicBool;
+use std::sync::Arc;
 
 #[derive(Clone)]
 enum ConnectionState {
     Connected,
 }
 
+#[derive(Clone)]
+pub enum SubscriptionState {
+    Subscribed,
+    UnSubscribed,
+}
+
+#[derive(Clone)]
 pub struct Subscription {
     status: ConnectionState,
-    pub subscription_id: u64,
+    pub subscription_id: SubscriptionId,
     pub interested_events: Vec<EventTags>,
     sink: Sink,
+    pub cold_state: SubscriptionState,
```

Comment:
cold_state -> subscription_state

```suggestion
    pub subscription_state: SubscriptionState,
```